gateway-messages/src/sp_impl.rs

```diff
@@ -38,6 +42,26 @@ pub struct SocketAddrV6 {
     pub port: u16,
 }
 
+/// Description of a device as reported as part of this SP's inventory.
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub struct DeviceDescription<'a> {
+    pub device: &'a str,
+    pub description: &'a str,
+    pub num_measurement_channels: u32,
+    pub presence: DevicePresence,
+}
+
+impl From<DeviceDescription<'_>> for crate::DeviceDescription {
```

Comment:
I find this somewhat confusing, since they both have the same name. Later on, you alias `crate::DeviceDescription` as `DeviceDescriptionHeader`. I think it would be useful to just call it `crate::DeviceDescriptionHeader` to begin with. That's really all it's used for, as it doesn't contain enough information unserialized to do anything with.

Reply:
Good call; renamed in 8468175